live/engine/markout.py

```
@@ -0,0 +1,115 @@
+        if cid not in mids_cache:
+            try:
+                m = fetch_pinned_market(cid, require_rewards=False)
+                if m:
+                    up_book = full_book(clob_host, m.up_token)
+                    dn_book = full_book(clob_host, m.down_token)
+                    bb_up, ba_up = up_book.get("best_bid"), up_book.get("best_ask")
+                    bb_dn, ba_dn = dn_book.get("best_bid"), dn_book.get("best_ask")
+                    mid_up = (bb_up + ba_up) / 2.0 if (bb_up and ba_up) else None
+                    mid_dn = (bb_dn + ba_dn) / 2.0 if (bb_dn and ba_dn) else None
+                    mids_cache[cid] = {"UP": mid_up, "DOWN": mid_dn}
+            except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)